src/test/java/me/dinowernli/grpc/prometheus/integration/MonitoringClientInterceptorIntegrationTest.java

```diff
@@ -53,126 +53,176 @@ public void tearDown() throws Throwable {
   @Test
   public void unaryRpcMetrics() throws Throwable {
     createClientStub(CHEAP_METRICS).sayHello(REQUEST, responseRecorder);
-    assertThat(extractMetricValue("grpc_client_started_total")).isWithin(0).of(1);
-    assertThat(findRecordedMetricOrThrow("grpc_client_msg_received_total").samples).isEmpty();
-    assertThat(findRecordedMetricOrThrow("grpc_client_msg_sent_total").samples).isEmpty();
+    assertThat(
+            findRecordedMetricOrThrow("grpc_client_started").samples.stream()
+                    .filter(s -> s.name.equals("grpc_client_started_total"))
+                    .findFirst().get().value)
+            .isWithin(0).of(1);
+
+    assertThat(findRecordedMetricOrThrow("grpc_client_msg_received").samples).isEmpty();
+    assertThat(findRecordedMetricOrThrow("grpc_client_msg_sent").samples).isEmpty();
 
     responseRecorder.awaitCompletion();
 
     Collector.MetricFamilySamples handled = findRecordedMetricOrThrow("grpc_client_completed");
-    assertThat(handled.samples).hasSize(1);
-    assertThat(handled.samples.get(0).labelValues).containsExactly(
-        "UNARY", HelloServiceImpl.SERVICE_NAME, HelloServiceImpl.UNARY_METHOD_NAME, 
-        "OK", "OK"); // TODO: These are the "code" and "grpc_code" labels which are currently duplicated. "code" should be deprecated in a future release.
-    assertThat(handled.samples.get(0).value).isWithin(0).of(1);
+    assertThat(handled.samples).hasSize(2);
+    Collector.MetricFamilySamples.Sample totalSample =
+            handled.samples.stream().filter(s -> s.name.equals("grpc_client_completed_total")).findFirst().get();
+    assertThat(totalSample.labelValues).containsExactly(
+            "UNARY", HelloServiceImpl.SERVICE_NAME, HelloServiceImpl.UNARY_METHOD_NAME,
+            "OK", "OK"); // TODO: These are the "code" and "grpc_code" labels which are currently duplicated. "code" should be deprecated in a future release.
+    assertThat(totalSample.value).isWithin(0).of(1);
   }
 
   @Test
   public void clientStreamRpcMetrics() throws Throwable {
     StreamObserver<HelloProto.HelloRequest> requestStream =
-        createClientStub(CHEAP_METRICS).sayHelloClientStream(responseRecorder);
+            createClientStub(CHEAP_METRICS).sayHelloClientStream(responseRecorder);
     requestStream.onNext(REQUEST);
     requestStream.onNext(REQUEST);
 
-    assertThat(extractMetricValue("grpc_client_started_total")).isWithin(0).of(1);
+    assertThat(
+            findRecordedMetricOrThrow("grpc_client_started").samples.stream()
+                    .filter(s -> s.name.equals("grpc_client_started_total"))
+                    .findFirst().get().value)
```

Comment:
nit: would this test file benefit from an additional helper that takes care of the whole filter > find > get > value chain? I think the tests might end up more readable then, and it's an opportunity to make the error messages more readable (if, say, findFirst fails).

Reply:
Sure, will I can do that in the next PR to avoid nasty conflicts.